src/pretix/static/pretixpresale/js/ui/main.js

```diff
@@ -263,6 +263,10 @@ function setup_basics(el) {
 
     el.find(".js-only").removeClass("js-only");
     el.find(".js-hidden").hide();
+    // make sure to always have a #content for skip-link to work
+    if (!el.find("#content").length) {
+        el.find("main").first().attr("id", "content");
```

Comment:
The comment says "always", but we do not have a main element on every page. For example, I found no #content on the customer account login page, the waiting list page, pages generated by pretix-exhibitor or pretix-pages, etc.

Either we need to add main elements everywhere, or hack in another fallback like this:

```
$(".page-header").first().next()
```

Reply:
Good catch again, the pages I checked all had `main`. I also stumbled about the „Best-practice“ recommendation in our a11y-report that every page should have a main landmark. So I guess, at some point we should add a main-landmark to all pages, but for a robust solution, we should use your suggestion as a fallback, when no main is found. Although currently for my install this jumps to an alert-warning, but that is actually a good idea to not skip an alert-warning. I will add this.

Reply:
I used `document.querySelector('main') || document.querySelector('.page-header + *')` as a fallback, if no `#content` is available in the document.